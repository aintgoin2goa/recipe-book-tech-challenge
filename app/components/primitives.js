import styled, { createGlobalStyle } from 'styled-components';
import { Link } from 'react-router-dom';

import {
    TEXT_COLOR,
    HEADING_COLOR,
    HIGHLIGHT_COLOR,
    TEXT_BG,
    HEADING_BG,
    CARD_BG,
    CARD_LABEL_COLOR,
} from '../styles/colors';

import {
    DEFAULT_FONT,
    H1_SIZE,
    H2_SIZE,
    BASE_FONT_SIZE,
} from '../styles/typography';

import {
    TEXT_PADDING,
    LIST_SPACING,
} from '../styles/measurements';

export const Body = createGlobalStyle`
    html, body{
        margin: 0;
        padding: 0;
    }

    body {
        font-family: ${DEFAULT_FONT};
        font-size: ${BASE_FONT_SIZE};
        color: ${TEXT_COLOR};
    }
`;

export const Title = styled.h1`
    font-size: ${H1_SIZE};
    color: ${HEADING_COLOR};
    background-color: ${HEADING_BG};
    margin: 0;
    padding: ${TEXT_PADDING};
`;

export const H2 = styled.h2`
    font-size: ${H2_SIZE};
    margin: 0;
`;

export const SummaryText = styled.p`
    margin: 0;
`;

export const NoRecipesText = styled.p`
    font-size: ${H1_SIZE};
    color: ${HIGHLIGHT_COLOR};
    padding: ${TEXT_PADDING};
`;

export const ListContainer = styled.div`
    display: grid;
    grid-template-columns: auto auto;
    grid-gap: ${LIST_SPACING};
    margin: ${LIST_SPACING};

    @media only screen and (min-width:600px) {
        grid-template-columns: auto auto auto;
    }

    @media only screen and (min-width: 1600px) {
        grid-template-columns: auto auto auto auto;
    }
`;

export const Card = styled.div`
    box-shadow: 10px 10px 23px -6px rgba(0,0,0,0.75);
    box-sizing: border-box;
    background-color: ${CARD_BG};
    background-size: cover;
    height: 200px;
    position:relative;
    transition: opacity 250ms linear;
    opacity: 1;

    &:hover {
        opacity: 0.8;
    }
`;

export const StyledLink = styled(Link)`
    position:absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    text-decoration: none;
`;

export const Label = styled.span`
    box-sizing: border-box;
    padding: 1em;
    width: 100%;
    background-color: ${TEXT_BG};
    position:absolute;
    left: 0;
    bottom: 0;
    color: ${CARD_LABEL_COLOR};
`;

export const RecipeDetailLayout = styled.div`
    display: grid;
    grid-template-columns: 3fr 1fr;
    grid-template-rows: 80px auto;
    grid-template-areas: 
        "header header"
        "...... text";
    background-size: cover;
    height: 90vh;
`;

export const RecipeHeader = styled.h1`
    font-size: 2em;
    color: ${TEXT_COLOR};
    grid-area: header;
    background-color: ${TEXT_BG};
    margin: 0;
    padding: ${TEXT_PADDING};
`;

export const RecipeText = styled.div`
   grid-area: text;
   background-color: ${TEXT_BG};
   padding: ${TEXT_PADDING};
`;