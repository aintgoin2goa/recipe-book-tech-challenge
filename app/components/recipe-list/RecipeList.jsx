import React from 'react';
import {observer} from 'mobx-react';
import styled from 'styled-components';

import RecipeSummary from '../recipe-summary/RecipeSummary';

const ListContainer = styled.div`
    display: grid;
    grid-template-columns: auto auto;
    grid-gap: 10px;

    @media only screen and (min-width:600px) {
        grid-template-columns: auto auto auto;
    }

    @media only screen and (min-width: 1600px) {
        grid-template-columns: auto auto auto auto;
    }
`;

@observer
export default class RecipeList extends React.Component {
    render() {
        return (
            <ListContainer>
                {this.props.store.recipes.map(r => <RecipeSummary key={r.id} {...r} />)}
            </ListContainer>
        )
    }
}