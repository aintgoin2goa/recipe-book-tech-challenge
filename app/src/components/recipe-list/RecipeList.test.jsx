import renderer from 'react-test-renderer';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import RecipeList from './RecipeList';
import stubData from '../../../../api/data/recipes.json';
import { ListContainer } from '../primitives';


test('it should render a list of recipes', () => {
    const fetchRecipes = jest.fn();
    const props = {
        recipeList: {
            fetchRecipes,
            recipes: stubData.recipes.map(r => Object.assign({}, r, {image: r.imageUrl}))
        }
    }

    const tree = renderer.create(<MemoryRouter><RecipeList {...props}/></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it should show a message if there are no recipes', () => {
    const expectedMessage = 'Sorry, we currently have no recipes for you';
    const fetchRecipes = jest.fn();
    const props = {
        recipeList: {
            fetchRecipes,
            recipes: []
        }
    }

    const tree = renderer.create(<MemoryRouter><RecipeList {...props}/></MemoryRouter>).toJSON();
    const message = tree.children[0].children[0];
    expect(message).toEqual(expectedMessage);
});