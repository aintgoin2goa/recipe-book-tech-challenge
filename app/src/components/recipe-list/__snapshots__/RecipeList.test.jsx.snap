// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render a list of recipes 1`] = `
<div
  className="sc-ifAKCX hSsEoE"
>
  <div
    className="sc-EHOje goAuOW"
    style={
      Object {
        "backgroundImage": "url('http://ichef.bbci.co.uk/food/ic/food_16x9_608/recipes/lemonchicken_11654_16x9.jpg')",
      }
    }
  >
    <a
      className="sc-bZQynM kBuUpE"
      href="/recipe/lemon-chicken"
      onClick={[Function]}
    >
      <span
        className="sc-gzVnrw cjozps"
      >
        <h2
          className="sc-bwzfXH bhIhtX"
        >
          Lemon Chicken
        </h2>
        <p
          className="sc-htpNat dNVVyN"
        >
          🕐 
          30 minutes
           
          <br />
          Chicken Breasts, Lemon, Thyme
        </p>
      </span>
    </a>
  </div>
  <div
    className="sc-EHOje goAuOW"
    style={
      Object {
        "backgroundImage": "url('http://ichef.bbci.co.uk/food/ic/food_16x9_608/recipes/beefstroganoffwithsa_89439_16x9.jpg')",
      }
    }
  >
    <a
      className="sc-bZQynM kBuUpE"
      href="/recipe/beef-stroganoff"
      onClick={[Function]}
    >
      <span
        className="sc-gzVnrw cjozps"
      >
        <h2
          className="sc-bwzfXH bhIhtX"
        >
          Beef Stroganoff
        </h2>
        <p
          className="sc-htpNat dNVVyN"
        >
          🕐 
          30 minutes
           
          <br />
          Beef, Mustard, Mushrooms
        </p>
      </span>
    </a>
  </div>
  <div
    className="sc-EHOje goAuOW"
    style={
      Object {
        "backgroundImage": "url('http://ichef.bbci.co.uk/food/ic/food_16x9_608/recipes/chickencaesarsalad_84099_16x9.jpg')",
      }
    }
  >
    <a
      className="sc-bZQynM kBuUpE"
      href="/recipe/chicken-ceasar-salad"
      onClick={[Function]}
    >
      <span
        className="sc-gzVnrw cjozps"
      >
        <h2
          className="sc-bwzfXH bhIhtX"
        >
          Chicken Caesar Salad
        </h2>
        <p
          className="sc-htpNat dNVVyN"
        >
          🕐 
          25 minutes
           
          <br />
          Lettuce, Chicken, Parmesan, Croutons
        </p>
      </span>
    </a>
  </div>
</div>
`;
